### PR TITLE
LibWeb: Implement `::file-selector-button` pseudo element

### PR DESCRIPTION
**Pull request by lukewarlow**
The button for a file input now matches `::file-selector-button`.

### CODE REVIEW

File: Libraries/LibWeb/CSS/Default.css
Comment (lukewarlow) [inline Libraries/LibWeb/CSS/Default.css:51]: For some reason the button lost all its styles just by matching the new pseudo element, so this fixes that.

**Comment by lukewarlow** [comment]
The below tests now pass as a result of this change:

- https://wpt.live/css/css-pseudo/file-selector-button-001.html
- https://wpt.live/css/css-pseudo/file-selector-button-inherit.html

The below test still fails:

- https://wpt.live/css/css-pseudo/file-selector-button-after-part.html - this relies on the element backed pseudo element concept being implemented. See https://drafts.csswg.org/css-pseudo/#element-like

File: Libraries/LibWeb/CSS/Default.css
Comment (AtkinsSJ) [reply Libraries/LibWeb/CSS/Default.css:51]: I can't decide if that's because of a bug, or correct. :sweat_smile: Shadow styles continue to confuse me. This is good for now anyway!

**Comment by AtkinsSJ** [comment]
Looks good, could you import those passing tests? Or otherwise include your own test for this, but importing should be easier.

File: Libraries/LibWeb/CSS/Default.css
Comment (Psychpsyo) [reply Libraries/LibWeb/CSS/Default.css:51]: > I can't decide if that's because of a bug, or correct. 😅 Shadow styles continue to confuse me. This is good for now anyway!

[Element-backed Pseudo-Elements](https://drafts.csswg.org/css-pseudo/#element-like) says that they "often are real elements that are not otherwise selectable".
Since `::file-selector-button` doesn't say anything about it violating that rule, I'd assume that this applies here.
Then again, I think I might be missing whatever spec actually defines the shadow tree stuff inside the input element. (if that even exists)

**Comment by lukewarlow** [comment]
@AtkinsSJ hopefully that's worked, the importer is missing functionality to handle one of these tests so I imported it manually. (About to make a PR to fix the importer)

**Comment by lukewarlow** [comment]
Raised an issue for the importer, because I think there might be a larger issue with mismatch tests.

File: Libraries/LibWeb/CSS/Default.css
Comment (lukewarlow) [reply Libraries/LibWeb/CSS/Default.css:51]: I would have expected this Default.css file to be loaded into Shadow DOM anyway. And the button would still match the element even if it has a pseudo attached to it.

I reckon there's a bug where an element has pseudo attached and that breaks it matching the raw element selector too or something.

Comment (lukewarlow) [reply Libraries/LibWeb/CSS/Default.css:51]: > Then again, I think I might be missing whatever spec actually defines the shadow tree stuff inside the input element. (if that even exists)

That's the fun part, there isn't a spec for it 😅 They're just vaguely defined in CSS. `::details-content` is at least defined properly now.

**Review by AtkinsSJ — APPROVED** [review]
From the CI logs:

```
Fail: Ref/input/wpt-import/css/css-pseudo/file-selector-button-001.html
```

That's because of #2712. You could skip the test, but I think it's worth waiting until #2754 is merged and then running CI again. Change otherwise looks fine.

**Comment by tcl3** [comment]
> I think it's worth waiting until #2754 is merged and then running CI again.

Note that after #2754 is merged, you'll need to use `WPT.sh` to reimport the test, as the importer needs to rewrite the link to the expectation file.

File: Tests/LibWeb/Text/expected/wpt-import/css/css-pseudo/file-selector-button-inherit.txt
Comment (tcl3) [inline Tests/LibWeb/Text/expected/wpt-import/css/css-pseudo/file-selector-button-inherit.txt:11]: This test needs to be rebaselined, as the expectation file format for imported WPT tests changed in #2714.

**Review by AtkinsSJ — APPROVED** [review]

**Pull request merged**
